### PR TITLE
feat(GuildManager): adds GuildManager#fetch

### PR DESCRIPTION
**Pull request by Fyko**
**Please describe the changes this PR makes and why it should be merged:**
Adds support for the **`GET`** `/guilds/:id` endpoint. API support! :^)

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/managers/GuildManager.js [hunk 1]
```diff
@@ -211,6 +211,25 @@ class GuildManager extends BaseManager {
         }, reject),
     );
   }
+
+  /**
+   * Obtains a guild from Discord, or the guild cache if it's already available.
+   * @param {Snowflake} id ID of the guild
+   * @param {boolean} [cache=true] Whether to cache the new guild object if it isn't already
+   * @returns {Promise<Channel>}
```
Comment (RDambrosio016) [inline src/managers/GuildManager.js:219]:
```suggestion
   * @returns {Promise<Guild>}
```
Copy and paste mistake? :)

File: src/managers/GuildManager.js [hunk 2]
```diff
@@ -211,6 +211,25 @@ class GuildManager extends BaseManager {
...
+   * @example
+   * // Fetch a channel by its id
+   * client.guilds.fetch('222078108977594368')
+   *   .then(channel => console.log(channel.name))
```
Comment (RDambrosio016) [inline src/managers/GuildManager.js]:
```suggestion
   *   .then(guild => console.log(guild.name))
```
same here

File: src/managers/GuildManager.js [hunk 3]
```diff
@@ -211,6 +211,25 @@ class GuildManager extends BaseManager {
...
     );
   }
+
+  /**
+   * Obtains a guild from Discord, or the guild cache if it's already available.
+   * @param {Snowflake} id ID of the guild
+   * @param {boolean} [cache=true] Whether to cache the new guild object if it isn't already
+   * @returns {Promise<Channel>}
+   * @example
+   * // Fetch a channel by its id
```
Comment (RDambrosio016) [inline src/managers/GuildManager.js]:
```suggestion
   * // Fetch a guild by its id
```
same here

File: src/managers/GuildManager.js [hunk 4]
```diff
@@ -211,6 +211,25 @@ class GuildManager extends BaseManager {
...
+   * client.guilds.fetch('222078108977594368')
+   *   .then(guild => console.log(guild.name))
+   *   .catch(console.error);
+   */
+  async fetch(id, cache = true) {
+    const existing = this.cache.get(id);
+    if (existing) return existing;
+
+    const data = await this.client.api.guilds(id).get();
+    return this.add(data, null, cache);
```
Comment (NotSugden) [inline src/managers/GuildManager.js:231]:
```suggestion
    return this.add(data, cache);
```
https://github.com/discordjs/discord.js/blob/2583ad5da7db5ab92dfa357909f9c713db0a2981/src/managers/BaseManager.js#L44

File: src/managers/GuildManager.js [hunk 5]
```diff
@@ -211,6 +211,25 @@ class GuildManager extends BaseManager {
...
+   * @param {Snowflake} id ID of the guild
+   * @param {boolean} [cache=true] Whether to cache the new guild object if it isn't already
+   * @returns {Promise<Guild>}
+   * @example
+   * // Fetch a guild by its id
+   * client.guilds.fetch('222078108977594368')
+   *   .then(guild => console.log(guild.name))
+   *   .catch(console.error);
+   */
+  async fetch(id, cache = true) {
```
Comment (vladfrangu) [inline src/managers/GuildManager.js:226]: This doesn't allow us to specify `with_counts` per https://discord.com/developers/docs/resources/guild#get-guild-query-string-params

Do we want to just default it to true?

Comment (Fyko) [reply src/managers/GuildManager.js]: Sounds good! I additionally added the force param from #4592

File: typings/index.d.ts [hunk 6]
```diff
@@ -1862,6 +1862,7 @@ declare module 'discord.js' {
       name: string,
       options?: { region?: string; icon: BufferResolvable | Base64Resolvable | null },
     ): Promise<Guild>;
+    public fetch(id: Snowflake, cache?: boolean): Promise<Guild>;
```
Comment (NotSugden) [inline typings/index.d.ts:1865]:
```suggestion
    public fetch(id: Snowflake, cache?: boolean, force?: boolean): Promise<Guild>;
```
force parameter

**Pull request merged**